src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -6,295 +6,192 @@
+    double[] xResult = new double[points.length];
+    double[] yResult = new double[points.length];
+    for (int i = 0; i < points.length; i++) {
+      xResult[i] = points[i].getX();
+      yResult[i] = points[i].getY();
+    }
+    return Tuple.of(xResult, yResult);
+  }
+
+  protected static double[] getCoords(PolynomialSplineFunction curve, double[] index) {
+    final double detailLevel = 1.0;//Tweak this value in order to achieve good rendering results
+    final double defaultCoordSize = index[index.length - 1];
+    final double[] coords = new double[(int)Math.round(detailLevel * defaultCoordSize) + 1];
+    final double stepSize = curve.getKnots()[curve.getKnots().length - 1] / coords.length;
+    double curValue = 0;
+    for (int i = 0; i < coords.length; i ++) {
+      coords[i] = curve.value(curValue);
+      curValue += stepSize;
+    }
+    return coords;
+  }
+
+  private static void drawMoveLength(Graphics2D graphics, RouteDescription routeDescription, Point[] points,
+      int xOffset, int yOffset, MapPanel view, boolean scrollWrapX, boolean scrollWrapY, int numTerritories,
+      String movementLeftForCurrentUnits) {
+    final double scale = view.getScale();
+    final Point cursorPos = points[points.length - 1];
+    final int xDir = cursorPos.x - points[numTerritories - 2].x;
+    final int textXOffset = xDir > 0 ? 6 : (xDir == 0 ? 0 : -14);
+
+    final int yDir = cursorPos.y - points[numTerritories - 2].y;
+    final int textYOffset = yDir > 0 ? 18 : -24;
+
+    final String unitMovementLeft =
+        movementLeftForCurrentUnits == null || movementLeftForCurrentUnits.trim().length() == 0 ? ""
+            : "    /" + movementLeftForCurrentUnits;
+    final BufferedImage movementImage = new BufferedImage(72, 24, BufferedImage.TYPE_INT_ARGB);
+    setupTextImage(movementImage.createGraphics(), String.valueOf(numTerritories - 1), unitMovementLeft);
+    graphics.drawImage(movementImage, (int) ((cursorPos.x + textXOffset - xOffset) * scale),
+        (int) ((cursorPos.y + textYOffset - yOffset) * scale), null);
+
+
+    final int translateX = scrollWrapX ? view.getImageWidth() : 0;
+    final int translateY = scrollWrapY ? view.getImageHeight() : 0;
+    graphics.drawImage(movementImage,
+        (int) ((cursorPos.x + textXOffset - xOffset - translateX) * scale),
+        (int) ((cursorPos.y + textYOffset - yOffset + translateY) * scale), null);
+    graphics.drawImage(movementImage,
+        (int) ((cursorPos.x + textXOffset - xOffset + translateX) * scale),
+        (int) ((cursorPos.y + textYOffset - yOffset - translateY) * scale), null);
+  }
+
+  private static void drawCurvedPath(Graphics2D graphics, Point[] points, MapPanel view) {
+    final double[] index = getIndex(points);
+    final Tuple<double[], double[]> pointArrays = pointsToDoubleArrays(points);
+    final PolynomialSplineFunction xcurve = splineInterpolator.interpolate(index, pointArrays.getFirst());
```

Comment:
Cohesion could be better here. IE: similar things grouped closely together.

IE:
```
xCurve = ...
xCoords = ...

yCurve = ...
yCoords = ...
```

In fact, it looks like the y coordinate stuff is not used/needed until line 183. So the y stuff could be moved down a few lines.